client/src/components/Chat.jsx

```diff
@@ -323,89 +320,145 @@ const Chat = () => {
 					className="h-[100%] md:max-h-full overflow-y-auto w-full scroll-smooth"
 				>
 					{sortedMessages.map(
-						({ senderId: sender, id, message, time, status, isEdited, oldMessages, containsBadword, isRead }) => {
+						({ senderId: sender, id, message, time, status, isEdited, oldMessages, containsBadword, isRead, replyTo }) => {
 							const isSender = sender.toString() === senderId.toString();
 							message = decryptMessage(message)
+							// original message this message is a reply to
+							const repliedMessage = replyTo ? (() => {
+								const messageObj = getMessage(replyTo)
+								if (!messageObj) {
+									return null
+								}
+
+								return {
+									...messageObj,
+									message: decryptMessage(messageObj.message)
+								}
+							})() : null
+
+							// is this message currently being replied? 
+							const hasActiveReply = currentReplyMessageId === id
 
 							return (
-								<div
-									key={id}
-									id={!isSender && `message-${id}`}
-									className={`w-full flex text-white relative mb-2 ${
-										isSender ? 'justify-end' : 'justify-start'
-									}`}
-								>
+								<div key={id} id={`message-${id}`} className={createClassesFromArray(
+									'flex flex-col gap-2 py-2 duration-500 transition-all',
+									hasActiveReply && 'bg-[#FF9F1C]/25 border-[#FF9F1C]',
+									hasActiveReply && (isSender ? 'border-r-[3.5px]' : 'border-l-[3.5px]')
+								)}>
+								{replyTo && (
 									<div
-										className={`flex flex-col mb-[2px] min-w-[10px] mdl:max-w-[80%] max-w-[50%] ${
-											isSender ? 'items-end' : 'items-start'
-										}`}
+										className={createClassesFromArray(
+											'max-w-[80%] md:max-w-[50%] min-w-[10px] flex gap-2 items-center',
+											sender.toString() === senderId.toString() && 'self-end',
+											repliedMessage && 'cursor-pointer'
+										)}
```

Comment:
here too, i'm trying to avoid using it

Reply:
Sorry, I don't get what you're talking about.